java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToVarHint.java

```diff
@@ -209,17 +228,18 @@ private static boolean isValidVarType(HintContext ctx) {
                 default:
                     break;
             }
+            // variable initializer type should be same as variable type.
+            TypeMirror initTypeMirror = ctx.getInfo().getTrees().getTypeMirror(initTreePath);
+            TypeMirror variableTypeMirror = ctx.getInfo().getTrees().getElement(treePath).asType();
+            if ((!Utilities.isValidType(initTypeMirror)) || (!ctx.getInfo().getTypes().isSameType(variableTypeMirror, Utilities.resolveCapturedType(ctx.getInfo(), initTypeMirror)))) {
+                return false;
+            }
+            return true;
+        } else if (expressionTreePath != null) {
+            ExecutableElement iterator = ExpandEnhancedForLoop.findIterable(ctx.getInfo());
+            return (iterator != null);
```

Comment:
Please add type check here as well, convert to var hint should show up if variable type is same as array or collection data type.

Reply:
@rtaneja1 , Thanks for pointing this out. I have fixed this in the latest commit and I have added a test for this as well.